spring-boot/proxy/src/main/java/eu/coatrack/proxy/metrics/MetricsCounterService.java

```diff
@@ -103,7 +112,7 @@ private Metric createMetricToTransmit(TemporaryMetricsAggregation tma) {
         metricToTransmit.setRequestMethod(tma.getRequestMethod());
         metricToTransmit.setType(tma.getMetricType());
         metricToTransmit.setHttpResponseCode(tma.getHttpResponseCode());
-        metricToTransmit.setDateOfApiCall(Date.valueOf(LocalDate.now()));
+        metricToTransmit.setDateOfApiCall(java.util.Date.from(now.atStartOfDay().toInstant(ZoneOffset.UTC)));
```

Comment:
This change seems to have no effect at all. Due to the UTC timezone set for the whole VM, the parameter `now` is already the current date according to UTC. The following would thus easier to read (also considering that the `now` parameter does not contain time information so maybe `today` seems to be a better name):

```
private Metric createMetricToTransmit(TemporaryMetricsAggregation tma, LocalDate today {
[...]
metricToTransmit.setDateOfApiCall(Date.valueOf(today));
```

Reply:
Ah sry, I think I delivered the wrong explanation.

In the previous approach, to create a metric we constructed two time objects with a little processing in between. When the times are created before and after midnight (which is a rare but possible case), this would result in a different dates and therefore inconsistencies. 

In the current approach a `LocalDate today` is created once an delivered as argument two times. Therefore, this inconsistency problem can't happen anymore.

So this "hacky" expression solves is the transformation of `LocalDate today` from the argument to a Date, see [here](https://beginnersbook.com/2017/10/java-convert-localdate-to-date/).

Reply:
After re-thinking the matter, I agree. Done as suggested.